Apps/APN-UIKit/APN UIKit/APN UIKit/View/Customisation/CustomDataViewController.swift

```diff
@@ -46,13 +46,30 @@ class CustomDataViewController: UIViewController {
             propertyValueLabel.text = "Attribute Value"
         }
     }
+    
+    func isAllTextFieldsValid() {
+        if propertyValueTextField.isTextTrimEmpty ||
+            propertyValueTextField.isTextTrimEmpty ||
+            (source == .customEvents && eventNameTextField.isTextTrimEmpty) {
+            showAlert(withMessage: "Please fill all fields", .error)
+            return
+        }
+    }
 
     // MARK: - Actions
     
     @IBAction func sendCustomData(_ sender: UIButton) {
+        isAllTextFieldsValid()
```

Comment:
```suggestion
        guard isAllTextFieldsValid() else { return }
```

I think we will want to return from `sendCustomData()` function early if the text fields do not have valid data in them?

Reply:
fixed in another PR. Thanks for bringing this to notice.